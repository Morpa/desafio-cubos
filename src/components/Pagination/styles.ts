import { shade } from 'polished'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  display: flex;
  align-self: center;
`
export const List = styled.ul`
  display: flex;
  align-items: center;

  ${media.lessThan('medium')`
    flex-wrap: wrap;
    padding: 0 16px;
  `}
`
export const Page = styled.li`
  ${({ theme }) => css`
    list-style: none;
    color: ${theme.colors.primary};
    margin: 0.4rem;
    transition: ${theme.transition.default};

    &:hover {
      color: ${shade(0.2, theme.colors.primary)};
      cursor: pointer;
    }
  `}
`
