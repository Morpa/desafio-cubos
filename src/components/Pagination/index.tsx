import Badge from 'components/Badge'

import * as S from './styles'

export type PaginationProps = {
  moviesPerPag: number
  totalMovies: number
  paginate: number
  currentPag: number
}

const Pagination = ({
  moviesPerPag,
  totalMovies,
  /* paginate, */
  currentPag
}: PaginationProps) => {
  const pagNumbers = []

  for (let i = 1; i <= Math.ceil(totalMovies / moviesPerPag); i++) {
    pagNumbers.push(i)
  }

  return (
    <S.Wrapper>
      <S.List>
        {pagNumbers.map((page) =>
          page === currentPag ? (
            <S.Page key={page}>
              <Badge size="small">{page}</Badge>
            </S.Page>
          ) : (
            <S.Page key={page}>
              <h2>{page}</h2>
            </S.Page>
          )
        )}
      </S.List>
    </S.Wrapper>
  )
}

export default Pagination
