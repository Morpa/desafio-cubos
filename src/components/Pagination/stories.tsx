import { Story, Meta } from '@storybook/react/types-6-0'
import Pagination, { PaginationProps } from '.'

export default {
  title: 'Pagination',
  component: Pagination,
  args: {
    moviesPerPag: 5,
    totalMovies: 25,
    paginate: 1,
    currentPag: 1
  }
} as Meta

export const Default: Story<PaginationProps> = (args) => (
  <Pagination {...args} />
)
