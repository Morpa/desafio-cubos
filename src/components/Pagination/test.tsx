import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Pagination from '.'

const props = {
  moviesPerPag: 5,
  totalMovies: 25,
  paginate: 1,
  currentPag: 1
}

describe('<Pagination />', () => {
  it('should render badge when current page is active', () => {
    const { container } = renderWithTheme(<Pagination {...props} />)

    expect(
      screen.queryByRole('heading', { name: /1/i })
    ).not.toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render quantity of pages', () => {
    const { container } = renderWithTheme(<Pagination {...props} />)

    expect(container.querySelectorAll('li')).toHaveLength(5)

    expect(container.firstChild).toMatchSnapshot()
  })
})
