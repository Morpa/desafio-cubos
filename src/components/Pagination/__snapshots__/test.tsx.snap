// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render badge when current page is active 1`] = `
.c3 {
  background: #116193;
  padding: 0.2rem;
  border-radius: 50%;
  color: #00ede7;
  position: relative;
  height: 4.0rem;
  width: 4.0rem;
}

.c3 .c4 {
  border: 0.2rem solid #00ede7;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  list-style: none;
  color: #116193;
  margin: 0.4rem;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c2:hover {
  color: #0d4d75;
  cursor: pointer;
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    padding: 0 16px;
  }
}

<div
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 c5"
        >
          1
        </div>
      </div>
    </li>
    <li
      class="c2"
    >
      <h2>
        2
      </h2>
    </li>
    <li
      class="c2"
    >
      <h2>
        3
      </h2>
    </li>
    <li
      class="c2"
    >
      <h2>
        4
      </h2>
    </li>
    <li
      class="c2"
    >
      <h2>
        5
      </h2>
    </li>
  </ul>
</div>
`;

exports[`<Pagination /> should render quantity of pages 1`] = `
.c3 {
  background: #116193;
  padding: 0.2rem;
  border-radius: 50%;
  color: #00ede7;
  position: relative;
  height: 4.0rem;
  width: 4.0rem;
}

.c3 .c4 {
  border: 0.2rem solid #00ede7;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  list-style: none;
  color: #116193;
  margin: 0.4rem;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c2:hover {
  color: #0d4d75;
  cursor: pointer;
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    padding: 0 16px;
  }
}

<div
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 c5"
        >
          1
        </div>
      </div>
    </li>
    <li
      class="c2"
    >
      <h2>
        2
      </h2>
    </li>
    <li
      class="c2"
    >
      <h2>
        3
      </h2>
    </li>
    <li
      class="c2"
    >
      <h2>
        4
      </h2>
    </li>
    <li
      class="c2"
    >
      <h2>
        5
      </h2>
    </li>
  </ul>
</div>
`;
